cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +794,27 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+int32_t gd_fn_instr_utf8(int64_t context, const char* string, int32_t string_len,
+                         const char* substring, int32_t substring_len) {
+  if (string_len <= 0 || substring_len <= 0) {
+    return 0;
+  }
+
+  std::string str(string, string_len);
```

Comment:
We should avoid using std::string as this does a memcpy. It would also do a malloc, malloc requires a lock impacting performance on multi-threaded machines.

You should write a loop like as follows:
for(int i = 0; i < (string_len - substring_len); i++) {
  if (string[i] == substring[0]) {
    // check if there is a complete match starting from this index
  }
}

Reply:
Ok, fixed it